lib/datadog/ci/contrib/rspec/example.rb

```diff
@@ -37,6 +37,8 @@ def run(example_group_instance, reporter)
                 },
                 service: configuration[:service_name]
               ) do |test_span|
+                test_span.set_parameters({}, {"scoped_id" => metadata[:scoped_id]})
```

Comment:
Is :scoped_id comparable to an index (eg: if you change the values of the parameters, the first value will still have the same `scoped_id`)?

I'd be slightly worried about having different parameter values with the same `scoped_id`, in theory, but I imagine that'd be more or less impossible to do without involving some kind of resource not covered by code coverage (which is a problem that exists regardless of parameters).

Reply:
Yes, if parameter value will change the `scoped_id` will stay the same.

So if you had tests like:

```ruby
[1, 2, 3, 4, 5] do |number|
  it "prints number" do
    expect(print(number)).to be_truthy
  end
end
```

and then you changed them like that:

```ruby
[1, 2, 4, 5] do |number|
  it "prints number" do
    expect(print(number)).to be_truthy
  end
end
```

Now test that tests parameter "4" has the same fingerprint as test that previously tested "3" and the test with "5" now the same that the test with "4" before.

The issue is that with dynamically generated code like that we don't even know that this example was created 5 times with different variables in scope, because there is no passing of argument to any method happening here per se. 

I am 99% sure though that there could be a way to catch it in Ruby (because almost everything is possible if you try hard enough), but right now I am not convinced that we need to invest a lot of time into that right now.

There is an easy way for app developers to distinguish these tests based on parameter:

```ruby
[1, 2, 4, 5] do |number|
  it "prints number #{number}" do
    expect(print(number)).to be_truthy
  end
end
```

Then we'll see tests "prints number 1", "prints number 2", etc in the dashboard.

Reply:
From ITR's perspective, I think that's okay (because you can't change the parameter value without modifying the file, so that'll mark it as modified).

I'm not sure what the impact is for some of our other tools (eg: flaky test detection, or some of the test history queries that power the UI), but I think you're likely right that going this route for now is the proper decision.